lib/spack/spack/util/web.py

```diff
@@ -177,7 +177,11 @@ def read_from_url(url, accept_content_type=None):
 
     # Do the real GET request when we know it's just HTML.
     req.get_method = lambda: "GET"
-    response = _urlopen(req, timeout=_timeout, context=context)
+
+    try:
+        response = _urlopen(req, timeout=_timeout, context=context)
+    except URLError as err:
+        raise SpackWebError(err)
```

Comment:
@opadron This is causing issues for Python 3.x but I think the general issue is that `SpackWebError` is initialized with a string message vs. an exception (see `spack.error.SpackError).